src/sources/kafka.rs

```diff
@@ -234,10 +238,15 @@ async fn kafka_source(
                 let payload = Cursor::new(Bytes::copy_from_slice(payload));
 
                 let mut stream = FramedRead::new(payload, decoder.clone());
+                let (count, _) = stream.size_hint();
                 let mut stream = stream! {
                     loop {
                         match stream.next().await {
-                            Some(Ok((events, _))) => {
+                            Some(Ok((events, byte_size))) => {
+                                emit!(&KafkaEventsReceived {
+                                    count: events.len(),
+                                    byte_size,
```

Comment:
This one is also wrong, it should be using `BytesSizeOf::size_of` to report the in-memory bytes.